ERCS/erc-7540.md

```diff
@@ -66,29 +65,49 @@ After submission, Requests go through Pending, Claimable, and Claimed stages. An
+| Claimable   |                                 | *Internal Request fulfillment*:  `pendingDepositRequest[msg.sender] -= assets`; `claimableDepositRequest[owner] += assets` |
+| Claimed     | `deposit(assets, receiver)`      | `claimableDepositRequest[msg.sender] -= assets`; `vault.balanceOf[receiver] += shares` |
+
+Note that `maxDeposit` increases and decreases in sync with `claimableDepositRequest`.
 
 An important Vault inequality is that following a Request(s), the cumulative requested quantity MUST be more than `pendingDepositRequest + maxDeposit - claimed`. The inequality may come from fees or other state transitions outside implemented by Vault logic such as cancellation of a Request, otherwise this would be a strict equality.
 
 Requests MUST NOT skip or otherwise short-circuit the Claim state. In other words, to initiate and claim a Request, a user MUST call both request* and the corresponding Claim function separately, even in the same block. Vaults MUST NOT "push" tokens onto the user after a Request, users MUST "pull" the tokens via the Claim function.
 
 For asynchronous Vaults, the exchange rate between `shares` and `assets` including fees and yield is up to the Vault implementation. In other words, pending redemption Requests MAY NOT be yield bearing and MAY NOT have a fixed exchange rate.
 
+### Request Ids
+The request ID (`rid`) of a request is returned by the corresponding `requestDeposit` and `requestRedeem` functions.
+
+Multiple requests may have the same `rid`, so a given Request is discriminated by both the `rid` and the `owner`. 
+
+Requests of the same `rid` MUST be fungible with each other (except in the special case `rid == 0` described below). I.e. all Requests with the same `rid` MUST transition from Pending to Claimable at the same time and receive the same exchange rate between `assets` and `shares`.
+
+If a Request becomes partially claimable, all requests of the same `rid` MUST become claimable at the same pro rata rate.
+
+There are no assumptions or requirements of requests with different `rid`. I.e. they MAY transition to Claimable at different times and exchange rates with no ordering or correlation enforced in any way.
+
+When `rid==0`, the Vault MUST use purely the `owner` to discriminate the request state. The Pending and Claimable state of multiple requests from the same `owner` would be aggregated. If a Vault returns `0` for the `rid` of any request, it MUST return `0` for all requests.
 
 ### Methods
 
 #### requestDeposit
 
-Transfers `assets` from `msg.sender` into the Vault and submits a Request for asynchronous `deposit/mint`. This places the Request in Pending state, with a corresponding increase in `pendingDepositRequest` for the amount `assets`.
+Transfers `assets` from `msg.sender` into the Vault and submits a Request for asynchronous `deposit/mint`. This places the Request in Pending state, with a corresponding increase in `pendingDepositRequest` for the amount `assets`. 
```

Comment:
```suggestion
Transfers `assets` from `msg.sender` into the Vault and submits a Request for asynchronous `deposit`. This places the Request in Pending state, with a corresponding increase in `pendingDepositRequest` for the amount `assets`. 
```

Should we remove the word mint as really we expect the deposit function to be called when claimable to avoid confusion.